x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_package_custom_extension/components/fleet_integration_event_filters_card.test.tsx

```diff
@@ -0,0 +1,97 @@
+import { PolicyData } from '../../../../../../../../common/endpoint/types';
+import { getFoundExceptionListItemSchemaMock } from '../../../../../../../../../lists/common/schemas/response/found_exception_list_item_schema.mock';
+
+const endpointGenerator = new EndpointDocGenerator('seed');
+
+describe('Fleet integration policy endpoint security event filters card', () => {
+  let render: () => Promise<ReturnType<AppContextTestRender['render']>>;
+  let renderResult: ReturnType<AppContextTestRender['render']>;
+  let history: AppContextTestRender['history'];
+  let mockedContext: AppContextTestRender;
+  let mockedApi: ReturnType<typeof eventFiltersListQueryHttpMock>;
+  let policy: PolicyData;
+
+  beforeEach(() => {
+    policy = endpointGenerator.generatePolicyPackagePolicy();
+    mockedContext = createAppRootMockRenderer();
+    mockedApi = eventFiltersListQueryHttpMock(mockedContext.coreStart.http);
+    ({ history } = mockedContext);
+    render = async () => {
+      await act(async () => {
+        renderResult = mockedContext.render(
+          <FleetIntegrationEventFiltersCard policyId={policy.id} />
+        );
+        await waitFor(mockedApi.responseProvider.eventFiltersList);
+      });
+      return renderResult;
+    };
+
+    history.push(getPolicyEventFiltersPath(policy.id));
+  });
+
+  afterEach(() => reactTestingLibrary.cleanup());
+
+  it('should call the API and render the card correctly', async () => {
+    mockedApi.responseProvider.eventFiltersList.mockReturnValue(
+      getFoundExceptionListItemSchemaMock(3)
+    );
+
+    await render();
+    expect(renderResult.getByTestId('eventFilters-fleet-integration-card')).toHaveTextContent(
+      'Event filters3'
+    );
+  });
+
+  it('should show the card even when no event filters associated with the policy', async () => {
+    mockedApi.responseProvider.eventFiltersList.mockReturnValue(
+      getFoundExceptionListItemSchemaMock(0)
+    );
+
+    await render();
+    expect(renderResult.getByTestId('eventFilters-fleet-integration-card')).toBeTruthy();
+  });
+
+  it('should have the correct manage event filters link', async () => {
+    mockedApi.responseProvider.eventFiltersList.mockReturnValue(
+      getFoundExceptionListItemSchemaMock(1)
+    );
+
+    await render();
+    expect(renderResult.getByTestId('eventFilters-link-to-exceptions')).toHaveAttribute(
+      'href',
+      `/app/security/administration/policy/${policy.id}/eventFilters`
+    );
+  });
+
+  it('should show an error toast when API request fails', async () => {
+    const errorMessage = 'Uh oh! API error!';
+    mockedApi.responseProvider.eventFiltersList.mockImplementation(() => {
+      throw new Error(errorMessage);
+    });
+
+    await waitFor(() => {
+      expect(mockedContext.coreStart.notifications.toasts.addDanger).toHaveBeenCalledWith(
+        `There was an error trying to fetch event filters stats: ${errorMessage}`
+      );
+    });
+  });
```

Comment:
@paul-tavares I can use some help with this particular test to pass. Not sure what I'm missing here.